src/naga_http/server.clj

```diff
@@ -4,15 +4,68 @@
   (:require [cheshire.core :as json]
             [compojure.core :refer :all]
             [compojure.route :as route]
-            naga.cli
-            [ring.middleware.defaults :refer [wrap-defaults api-defaults]]))
+            [naga.lang.pabu :as pabu]
+            [naga.engine :as e]
+            [naga.rules :as r]
+            [naga.store :as store]
+            [naga.data :as data]
+            [ring.middleware.defaults :refer [wrap-defaults api-defaults]])
+  (:import [java.util Date]))
+
+(def plain-headers {"Content-Type" "text/plain"})
+
+(def json-headers {"Content-Type" "application/json"})
+
+(def programs (atom {}))
+
+(defn uuid-str []
+  (-> (java.util.UUID/randomUUID)
+      .toString))
+
+(defn parse-program [text]
+  (let [{:keys [rules]} (pabu/read-str text)]
```

Comment:
I don't know if kibitz will respond to this, but when it's not a list of keys, I'm inclined to extract them directly.
i.e.
```clojure
(let [{rules :rules} (pabu/read-str text)]
```